repos/backend/libs/fileSys/testFiles.js

```diff
@@ -49,20 +49,31 @@ const getTestFile = async (config, testPath) => {
   } 
 }
 
+/**
+ * Save file at a given location. file should be located in the test root path
+ * @param {Object} config 
+ * @param {string} fullPath
+ * @param {string} content 
+ */
+const saveTestFile = async (config, fullPath, content) => {
 
-const saveTestFile = async (config, testPath, content) => {
   const { testsRoot } = config.paths
-  const fullPath = path.join(testsRoot, testPath)
+  const inTestRoot = fullPath.startsWith(testsRoot)
```

Comment:
This is a good check, but we should add further validation to this.

We should clean the string by removing special characters, except for '/'
  * Do a google search and check what characters are valid in a file name for Linux systems
  * We definitely want to remove ';' '&&' and ':', but I'm sure there's more

Reply:
Just wanted to get a full understanding of this edge case.
in what use scenario would this be an issue that the `writeFile` wouldnt catch if it fails?
if a user wants to save a file as `&&.js` for whatever reason shouldn't just let the system decides if that's allowed or not?

Reply:
Let's say your on a windows machine. You mount in your test folder

Then you run keg-herkin
Create a new test file, and call it 'awesome:tests.feature'

You write your tests, and everything is working fine. Nothing to worry about. Finish it up and call it a day and go eat a burro.

Tomorrow comes like most days, and you are reviewing your tests on your local file system.

For some reason, you can't seem to open 'awesome:tests.feature' locally. "WTF!", you say out loud.

So you start up keg-herkin, and the test loads fine. 

"WTF!", you say out loud **again**.

So what happened?

A quick google search will most likely tell you => https://www.google.com/search?client=safari&hl=en-us&ei=jro1YMfjEPPi9AOd0Ke4DA&q=windows+invalid+files+names&oq=windows+invalid+files+names&gs_lcp=ChNtb2JpbGUtZ3dzLXdpei1zZXJwEAMyBwgAEEcQsAMyBwgAEEcQsAMyBwgAEEcQsAMyBwgAEEcQsAMyBwgAEEcQsAMyBwgAEEcQsAMyBwgAEEcQsAMyBwgAEEcQsANQAFgAYL4VaAFwAHgAgAEAiAEAkgEAmAEAyAEIwAEB&sclient=mobile-gws-wiz-serp

Then to fix it, you install https://github.com/sindresorhus/filenamify

Later you see Lance, and you thank him for being awesome.

Reply:
best user story ever.
updated